variables.tf

```diff
@@ -66,59 +40,101 @@ variable "storage_class_name" {
 variable "prometheus_resources" {
   type        = any
   description = "Prometheus resource"
-  default     = { 
-                  cpu_requests = "", 
-                  memory_requests = "" 
+  default     = {
+                  cpu_requests = "",
+                  memory_requests = ""
                 }
 }
 
 variable "prometheus_operator_resources" {
   type        = any
   description = "PrometheusOperetor resource"
-  default     = { 
-                  cpu_requests = "", 
-                  memory_requests = "" 
-                }
-}
-
-variable "grafana_resources" {
-  type        = any
-  description = "Grafana resource"
-  default     = { cpu_requests = "", 
-                  memory_requests = "" 
+  default     = {
+                  cpu_requests = "",
+                  memory_requests = ""
                 }
 }
 
 variable "prometheus_adapter_resources" {
   type        = any
   description = "PrometheusAdapter resource"
-  default     = { 
-                  cpu_requests = "", 
-                  memory_requests = "" 
+  default     = {
+                  cpu_requests = "",
+                  memory_requests = ""
                 }
 }
 
-variable "company" {
-  type = string
-  default = "Company"
-}
-
-variable "ingress_class" {
-  type = string
-  default = "nginx"
-}
-
 variable "kube_etcd" {
   type = bool
+  description = "Component scraping etcd"
   default = false
 }
 
 variable "kube_controller_manager" {
   type = bool
+  description = "Component scraping the kube controller manager"
   default = false
 }
 
 variable "kube_scheduler" {
   type = bool
+  description = "Component scraping kube scheduler"
   default = false
-}
\ No newline at end of file
+}
+
+# Grafana config.
+variable "grafana_ingress_class" {
+  type = string
+  description = "Ingress Class"
+  default = "nginx"
+}
+
+variable "grafana_ingress_host" {
+  type = string
+  description = "Grafana ingress host"
+}
+
+variable "grafana_ingress_enabled" {
+  type = bool
+  description = "Enable Ingress for Grafana"
+  default = true
+}
+
+variable "grafana_ingress_basic_auth_username" {
+  type = string
+  description = "Grafana basic auth username (if not empty it will be activated)."
+  default = "admin"
+}
+
+variable "grafana_ingress_basic_auth_message" {
+  type = string
+  description = "Grafana basic auth message."
+  default = "Authentication Required"
+}
+
+variable "grafana_cert_manager_cluster_issuer_name" {
+  type = string
+  description = "Resource representing the cluster issuer of cert-manager (used to deploy a TLS cert for Grafana ingress)."
+  default = ""
```

Comment:
I think that we should not define a default. Without a default the error will be raised for `not defined variable` and not when the helm provider will try to install the chart.

Maybe we can be more explicit about the behaviour of this variable: if this variable is left empty the `cert-manager` annotations will not be put in place in the `Ingress`.